### PR TITLE
Avoid silent errors

### PR DESCRIPTION
**Pull request by suhaibmujahid**
With this PR, I was able to reproduce the `504` error using the example provided by @Archaeopteryx in #189:

```
Traceback (most recent call last):
  File "./test.py", line 68, in <module>
    data = my_function("test", start_date, end_date)
  File "./test.py", line 57, in my_function
    Bugzilla(
  File "/repos/mozilla/libmozdata/./libmozdata/bugzilla.py", line 177, in wait
    self.wait_bugs()
  File "/repos/mozilla/libmozdata/./libmozdata/bugzilla.py", line 188, in wait_bugs
    r.result()
  File "/opt/homebrew/Cellar/python@3.9/3.9.10/Frameworks/Python.framework/Versions/3.9/lib/python3.9/concurrent/futures/_base.py", line 446, in result
    return self.__get_result()
  File "/opt/homebrew/Cellar/python@3.9/3.9.10/Frameworks/Python.framework/Versions/3.9/lib/python3.9/concurrent/futures/_base.py", line 391, in __get_result
    raise self._exception
  File "/opt/homebrew/Cellar/python@3.9/3.9.10/Frameworks/Python.framework/Versions/3.9/lib/python3.9/concurrent/futures/thread.py", line 58, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/repos/mozilla/libmozdata/venv/lib/python3.9/site-packages/requests/sessions.py", line 529, in request
    resp = self.send(prep, **send_kwargs)
  File "/repos/mozilla/libmozdata/venv/lib/python3.9/site-packages/requests/sessions.py", line 652, in send
    r = dispatch_hook('response', hooks, r, **kwargs)
  File "/repos/mozilla/libmozdata/venv/lib/python3.9/site-packages/requests/hooks.py", line 31, in dispatch_hook
    _hook_data = hook(hook_data, **kwargs)
  File "/repos/mozilla/libmozdata/./libmozdata/bugzilla.py", line 556, in __bugs_cb
    res.raise_for_status()
  File "/repos/mozilla/libmozdata/venv/lib/python3.9/site-packages/requests/models.py", line 960, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 504 Server Error: Gateway Time-out for url: https://bugzilla.mozilla.org/rest/bug?include_fields=id&include_fields=product&include_fields=component&include_fields=severity&include_fields=history&product=Core&product=Firefox&product=Toolkit&f1=bug_group&o1=notsubstring&v1=security&f2=keywords&o2=nowords&v2=crash&f3=OP&j3=OR&f4=OP&j4=AND_G&f5=bug_severity&o5=changedfrom&v5=S1&f6=bug_severity&o6=changedafter&f7=bug_severity&o7=changedbefore&f9=CP&f10=OP&j10=AND_G&f11=bug_severity&o11=changedfrom&v11=S2&f12=bug_severity&o12=changedafter&f13=bug_severity&o13=changedbefore&f15=CP&f16=CP&v6=2010-01-01&v7=2022-01-31&v12=2010-01-01&v13=2022-01-31&limit=100&order=bug_id&offset=0
```

Considering what @marco-c  mentioned in #186, I made the default configurations to raise errors on failures. It can be turned off if needed.

### CODE REVIEW

**Review by marco-c — APPROVED** [review]

**Comment by marco-c** [comment]
It looks like there is a test failure

**Comment by suhaibmujahid** [comment]
An existing test now is failing the CI because of a  `500` error:
```
Traceback (most recent call last):
  File "/usr/local/lib/python3.9/site-packages/responses/__init__.py", line 180, in wrapper
    return func(*args, **kwargs)
  File "/libmozdata/tests/test_patchanalysis.py", line 409, in test_bug_analysis
    info = patchanalysis.bug_analysis(1007402)
  File "/libmozdata/libmozdata/patchanalysis.py", line 722, in bug_analysis
    patch_analysis(
  File "/libmozdata/libmozdata/patchanalysis.py", line 345, in patch_analysis
    info["code_churn_overall"] += len(hi.get(path, utc_ts_to=utc_ts_to)["patches"])
  File "/libmozdata/libmozdata/HGFileInfo.py", line 51, in get
    result.wait()
  File "/libmozdata/libmozdata/connection.py", line 134, in wait
    r.result()
  File "/usr/local/lib/python3.9/concurrent/futures/_base.py", line 446, in result
    return self.__get_result()
  File "/usr/local/lib/python3.9/concurrent/futures/_base.py", line 391, in __get_result
    raise self._exception
  File "/usr/local/lib/python3.9/concurrent/futures/thread.py", line 58, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/usr/local/lib/python3.9/site-packages/requests/sessions.py", line 529, in request
    resp = self.send(prep, **send_kwargs)
  File "/usr/local/lib/python3.9/site-packages/requests/sessions.py", line 652, in send
    r = dispatch_hook('response', hooks, r, **kwargs)
  File "/usr/local/lib/python3.9/site-packages/requests/hooks.py", line 31, in dispatch_hook
    _hook_data = hook(hook_data, **kwargs)
  File "/libmozdata/libmozdata/connection.py", line 123, in cb
    res.raise_for_status()
  File "/usr/local/lib/python3.9/site-packages/requests/models.py", line 960, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 500 Server Error: Internal Server Error for url: https://hg.mozilla.org/mozilla-central/json-filelog?node=default&file=b2g%2Fcomponents%2FWebappManager.jsm&revcount=4096
```

The [request URL](https://hg.mozilla.org/mozilla-central/json-filelog?node=default&file=b2g%2Fcomponents%2FWebappManager.jsm&revcount=4096) are truly given a 500 response if you visit:
image

There are some tests that do not expect errors. I will fix the tests.

File: tests/test_patchanalysis.py [hunk 1]
```diff
@@ -11,6 +11,7 @@
```
Comment (marco-c) [inline tests/test_patchanalysis.py:11]: This reminds me that patchanalysis is not being used by anyone AFAIK, we should maybe remove it at this point. It's likely very outdated.

**Review by marco-c — APPROVED** [review]
LGTM!

**Pull request merged**